string_manipulation.rb

```diff
@@ -1,30 +1,104 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  # swap each element
+  # move an index
+  i = 0
+  n = my_string.length
+  while i < n/2
+    first = my_string[i]
+    my_string[i] = my_string[n-i-1]
+    my_string[n-i-1] = first
+    i += 1
+  end
 end
 
 # A method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+  # iterate through each characters
+  # if space leave alone
+  # create stack of letters
+  # pop off onto string
+  n = my_words.length
+  i = 0
+  # CAN_ 0123 NAC
+  while i < n
+    word = []
+    j = i
+    while my_words[j] != " " && j < n
+      word << my_words[j]
+      j += 1
+    end
+    while i < j
+      my_words[i] = word.pop
+      i += 1
+    end
+    i += 1
+  end
 end
 
 # A method to reverse the words in a sentence, in place.
 def reverse_sentence(my_sentence)
-  puts "NOT IMPLEMENTED"
+  string_reverse(my_sentence)
+  reverse_words(my_sentence)
 end
 
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  j = my_phrase.length - 1
+  while i < j
+    if my_phrase[i] == " "

```

Comment:
Minor optimization: I'd recommend making this while instead of if, to account for multiple white spaces.